sklearn/utils/estimator_checks.py

```diff
@@ -847,6 +858,139 @@ def _generate_sparse_matrix(X_csr):
         yield sparse_format + "_64", X
 
 
+def check_array_api_input(name, estimator_orig, *, array_namespace):
+    """Check that the array_api Array gives the same results as ndarrays."""
+    xp = pytest.importorskip(array_namespace)
+
+    X, y = make_classification(random_state=42)
```

Comment:
don't we use `load_iris` here instead?

Reply:
I don't understand the question.

Reply:
the rest of the common tests use the iris dataset, we don't seem to use `make_classification` anywhere else. Here we could use iris too.

Reply:
Since we repeat that everywhere, I wanted to see the impact in terms of computational resources and both options are cheap enough.

```python
In [1]: from sklearn.datasets import load_iris

In [2]: %timeit load_iris()
215 µs ± 1.96 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [3]: from sklearn.datasets import make_classification

In [4]: %timeit make_classification(random_state=42)
164 µs ± 721 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

We could switch to `load_iris` for the sake of consistency but no strong opinion.